SimplenoteUITests/Generic.swift

```diff
@@ -41,41 +41,43 @@ func getToAllNotes() {
 
 class Table {
 
+    class func getAllCells() -> XCUIElementQuery {
+        return app.tables.element.children(matching: .cell)
+    }
+
     class func getVisibleLabelledCellsNumber() -> Int {
         // We need only the table cells that have X = 0 and a non-empty label
         // otherwise we will include invisible elements from Sidebar pane, when Notes List is open
         // or the elements from Notes List when Settings are open
         // or the visible tags search suggestions
-        let cellsNum = app.tables.element.children(matching: .cell).count
-        var matchesNum: Int = 0
-        guard cellsNum > 0 else { return matchesNum }
+        let cells = Table.getAllCells()
+        var matches = 0
+        guard cells.count > 0 else { return matches }
```

Comment:
Nitpick

```suggestion
        guard cells.isEmpty == false else { return matches }
```

From a performance point of view, using `isEmpty` is better than counting all the elements when the only thing we care about is whether the array is empty or not.

Reply:
@mokagio After replacing loops with collections manipulations, as you suggested 🔥, the number of `count == 0` and `count > 0` usages went down a lot. But two usages still remain, and in both cases the `count` was used on `XCUIElementQuery`, which is not an array, but a query, as I understood, and has no `isEmpty` value. 

I started looking more into this, and saw that it becomes possible once I convert the `XCUIElementQuery` to array with `compactMap { $0 }`, for example. But, I guess, this operation involves traversing through all elements too, which makes the whole thing meaningless?

Also, I've read more, and [found](https://betterprogramming.pub/check-swift-array-is-empty-f15e63bf1026) that `isEmpty` is faster than `count` only for Strings, but for Array the performance is almost identical, because Array conforms to [RandomAccessCollection](https://developer.apple.com/documentation/swift/randomaccesscollection).

I agree that `isEmpty` reads better, and can think of adding an `isEmpty()` extension `XCUIElementQuery`, which returns `count == 0`, but this looks not very honest, and only improves the look, but not the performance.

I'm not sure which way to go with this issue, and would appreciate your thoughts, as always.

I will leave this change for the future and meanwhile will merge this PR, with all your other suggestions 🚀  applied.